src/fuzz/protobufs/utils/spanner_emulator_ddl_statement_proto_to_string.cc

```diff
@@ -0,0 +1,188 @@
+        case ColumnDataInfo::BOOL:
+            return "BOOL";
+        case ColumnDataInfo::INT64:
+            return "INT64";
+        case ColumnDataInfo::FLOAT64:
+            return "FLOAT64";
+        case ColumnDataInfo::STRING:
+            switch (lengthType) {
+                case ColumnDataInfo::BOUND:
+                    // ensures length is a positive number
+                    length = std::abs(length) + 1;
+                    return absl::Substitute("STRING( $0 )", 
+                        std::to_string(length % 2621440));
+                case ColumnDataInfo::UNBOUND:
+                    return absl::Substitute("STRING( $0 )", 
+                        std::to_string(length));
+                case ColumnDataInfo::MAX:
+                    return "STRING( MAX )";
+                default:
+                    return ""; // never triggers
+            }
+        case ColumnDataInfo::BYTES:
+            switch (lengthType) {
+                case ColumnDataInfo::BOUND:
+                    // ensures length is a positive number
+                    length = std::abs(length) + 1;
+                    return absl::Substitute("BYTES( $0 )", 
+                        std::to_string(length % 10485760));
+                case ColumnDataInfo::UNBOUND:
+                    return absl::Substitute("BYTES( $0 )", 
+                        std::to_string(length));
+                case ColumnDataInfo::MAX:
+                    return "BYTES( MAX )";
+                default:
+                    return ""; // never triggers
+            }
+        case ColumnDataInfo::DATE:
+            return "DATE";
+        case ColumnDataInfo::TIMESTAMP:
+            return "TIMESTAMP";
+        default: // never occurs given the protobuf structure
+            return "";
+    }
+}
+
+// if this column should be not null, return the appropriate string
+std::string isColumnNotNullToString(bool isNotNull) {
+    return isNotNull ? "NOT NULL" : "";
+}
+
+// return the options for this column as a string
+// the current api only allows for timestamps
+std::string columnOptionsToString(bool allowCommitTimestamps) {
+    std::string allowTimestamps = allowCommitTimestamps ? "true" : "null";
+    return absl::Substitute("{ OPTIONS ( allow_commit_timestamp = { $0 } ) }",
+         allowTimestamps);
+}
+
+// generates a list of the primary keys' column names and their orientations
+std::string toPrimaryKeys(const google::protobuf::RepeatedPtrField<Column>& columns) {
+    std::string primaryKeysAsString = "";
+    for (int i = 0; i < columns.size(); i++) {
+        const Column& currCol = columns.at(i);
+        absl::StrAppend(&primaryKeysAsString, columnToPrimaryKey(currCol));
+    }
+    // columnToPrimaryKey(currCol) will have an extra comma at the end
+    primaryKeysAsString = primaryKeysAsString.substr(0, 
+        primaryKeysAsString.size() - 1);
+
+    return primaryKeysAsString;
+}
+
+std::string columnToPrimaryKey(Column column) {
```

Comment:
We may eventually want the option to leave orientation out.

I also think we don't want the brackets around the orientation--I think it's just "primary_key_name ASC, primary_key2_name DESC, ..."?

Reply:
Is the recommended solution to make orientation an optional field in the proto and add an extra check if it exists when parsing the proto?

Reply:
Keeping this as a TODO